### PR TITLE
Copy nullability of non-nullable members

### PR DESCRIPTION
**Pull request by cston**
`InheritNullableStateOfMember` was skipping members with non-nullable reference types.

### CODE REVIEW

**Comment by cston** [comment]
@dotnet/roslyn-compiler please review.

File: src/Compilers/CSharp/Test/Semantic/Semantics/NullableReferenceTypesTests.cs [hunk 1]
```diff
@@ -56769,6 +56769,80 @@ static void F(object? x)
...
+class C<T>
+{
+    internal T F;
+}
+class Program
+{
+    static void F0(string? s)
+    {
+        var a1 = new C<string>() { F = s };
+        _ = a1.F/*T:string?*/;
```
Comment (AlekseyTs) [inline src/Compilers/CSharp/Test/Semantic/Semantics/NullableReferenceTypesTests.cs]: >_ = a1.F/*T:string?*/; [](start = 8, length = 22)

Please add a an attempt to use this field in an unsafe manner. The goal of the fix is to modify observable behavior of the compiler.

File: src/Compilers/CSharp/Test/Semantic/Semantics/NullableReferenceTypesTests.cs [hunk 2]
```diff
@@ -56769,6 +56769,80 @@ static void F(object? x)
...
+        var b1 = a1;
+        _ = b1.F/*T:string!*/;
```
Comment (AlekseyTs) [inline src/Compilers/CSharp/Test/Semantic/Semantics/NullableReferenceTypesTests.cs]: >_ = b1.F/*T:string!*/; [](start = 8, length = 22)

This feels unexpected. I would expect nullability of fields to flow through an assignment. Please add a an attempt to use this field in an unsafe manner.

File: src/Compilers/CSharp/Test/Semantic/Semantics/NullableReferenceTypesTests.cs [hunk 3]
```diff
@@ -56769,6 +56769,80 @@ static void F(object? x)
...
+struct S<T>
+{
+    internal T F;
+}
+class Program
+{
+    static void F0(string? s)
+    {
+        var a1 = new S<string>() { F = s };
+        _ = a1.F/*T:string?*/;
```
Comment (AlekseyTs) [inline src/Compilers/CSharp/Test/Semantic/Semantics/NullableReferenceTypesTests.cs]: >a1.F/*T:string?*/; [](start = 12, length = 18)

Please add an attempt to use this field in an unsafe manner.

File: src/Compilers/CSharp/Test/Semantic/Semantics/NullableReferenceTypesTests.cs [hunk 4]
```diff
@@ -56769,6 +56769,80 @@ static void F(object? x)
...
+        var b1 = a1;
+        _ = b1.F/*T:string?*/;
```
Comment (AlekseyTs) [inline src/Compilers/CSharp/Test/Semantic/Semantics/NullableReferenceTypesTests.cs]: >_ = b1.F/*T:string?*/; [](start = 8, length = 22)

Please add an attempt to use this field in an unsafe manner.

**Comment by AlekseyTs** [comment]
Done with review pass (iteration 2)

File: src/Compilers/CSharp/Test/Semantic/Semantics/NullableReferenceTypesTests.cs (hunk 2 above)
Comment (cston) [reply src/Compilers/CSharp/Test/Semantic/Semantics/NullableReferenceTypesTests.cs]: I agree, nullability of class members should be copied. Logged [https://github.com/dotnet/roslyn/issues/31395](https://github.com/dotnet/roslyn/issues/31395).

---
In reply to: [236795755](https://github.com/dotnet/roslyn/pull/31392#discussion_r236795755) [](ancestors = 236795755)

**Pull request merged**